### PR TITLE
[NCL-7207] Build Config create/edit page - repository section

### PR DESCRIPTION
**Pull request by patrikk0123**

### CODE REVIEW

File: src/components/BuildConfigCreateEditPage/BuildConfigCreateEditPage.tsx [hunk 1]
```diff
@@ -387,6 +406,113 @@ export const BuildConfigCreateEditPage = ({ isEditPage = false }: IBuildConfigCr
...
 
+      <ContentBox padding marginBottom>
+        <Form
+          onSubmit={(e) => {
+            e.preventDefault();
+          }}
+        >
+          <FormGroup
+            isRequired
+            label={`${buildConfigEntityAttributes.scmRepository.title} URL`}
```
Comment (matedo1) [inline src/components/BuildConfigCreateEditPage/BuildConfigCreateEditPage.tsx:417]: Please share as much as possible content like this with SCM Repositories create page - for example all title and tooltip attributes. 

The same as valid for other common parts like validation, "blue box" displaying information whether repository is already synced or not (let's export it as separated component). Please create follow-up jiras to apply required changes on SCM Repositories create page.

The goal is to this section and SCM Repositories create page synced as much as possible.

File: src/components/BuildConfigCreateEditPage/BuildConfigCreateEditPage.tsx [hunk 2]
```diff
@@ -387,6 +406,113 @@ export const BuildConfigCreateEditPage = ({ isEditPage = false }: IBuildConfigCr
...
+              render={({ value, onChange, onBlur }) => (
+                <Switch
+                  id={scmRepositoryEntityAttributes.preBuildSyncEnabled.id}
+                  name={scmRepositoryEntityAttributes.preBuildSyncEnabled.id}
+                  label="Enabled"
+                  labelOff="Disabled"
+                  isChecked={selectedScmRepository?.preBuildSyncEnabled || value}
+                  onChange={onChange}
+                  onBlur={onBlur}
+                  isDisabled={!!selectedScmRepository}
```
Comment (matedo1) [inline src/components/BuildConfigCreateEditPage/BuildConfigCreateEditPage.tsx:480]: Is it possible to display the reason why it's disabled?

**Review by matedo1 — CHANGES_REQUESTED** [review]

**Review by matedo1 — APPROVED** [review]
Did you create new jira to integrate `ScmRepositoryUrlAlert` on SCM Repositories create page?

**Comment by patrikk0123** [comment]
> ScmRepositoryUrlAlert

NCL-8383

**Pull request merged**